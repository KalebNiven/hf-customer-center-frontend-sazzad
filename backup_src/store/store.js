import { createStore, applyMiddleware, compose } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import createSagaMiddleware from "redux-saga";
import reducer from "./reducer";
import rootSaga from "./saga";
import logger from "redux-logger";

// create the saga middleware
const sagaMiddleware = createSagaMiddleware();

// initial state
const initialState = {}

// add all middleware here
const middleware = [sagaMiddleware]

// devtools setup
const devTools = process.env.NODE_ENV === 'production' ? applyMiddleware(...middleware) : composeWithDevTools(applyMiddleware(...middleware, logger))

// create the store with middleware
const store = createStore(reducer, initialState, devTools);

// export store
export default store;

// run saga
sagaMiddleware.run(rootSaga);