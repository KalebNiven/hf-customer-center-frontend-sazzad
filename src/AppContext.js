import React, { createContext, useContext, useState, useEffect } from "react";

export const AppContext = createContext();

export const AppContextProvider = ({ children }) => {
  const [drawerOpen, setDrawerOpen] = useState(false);
  const [externalSiteModal, setExternalSiteModal] = useState({
    isVisible: false,
    link: "#",
    target: "_blank",
    segmentPageCategory: null,
    segmentTitle: null,
    segmentTargetMemberId: null,
    membershipKey: null,
    label: null,
  });
  const [innerWidth, setInnerWidth] = useState("");
  const [toastOpen, setToastOpen] = useState(false);
  const [toastState, setToastState] = useState("");
  const [planName, setPlanName] = useState("");
  const [openPaperLess, setOpenPaperLess] = useState(false);
  const [globalError, setGlobalError] = useState(false);
  const [acknowledgmentModal, setAcknowledgmentModal] = useState({
    isVisible: false,
  });

  const resetExternalModal = () => {
    setExternalSiteModal({
      isVisible: false,
      link: null,
      target: null,
      pageCategory: null,
      segmentPageCategory: null,
      segmentTitle: null,
      segmentTargetMemberId: null,
      membershipKey: null,
      label: null,
    });
  };

  const resetAcknowledgmentModal = () => {
    setAcknowledgmentModal({ isVisible: false });
  };
  /* Window Resize Listener */
  const onWindowResize = () => {
    const innerWidth = window.innerWidth;
    setInnerWidth(innerWidth);
  };

  useEffect(() => {
    window.addEventListener("resize", onWindowResize);
    const innerWidth = window.innerWidth;
    setInnerWidth(innerWidth);

    return () => {
      window.removeEventListener("resize", onWindowResize);
    };
  }, []);

  return (
    <AppContext.Provider
      value={{
        externalSiteModal,
        setExternalSiteModal,
        openPaperLess,
        setOpenPaperLess,
        toastState,
        setToastState,
        planName,
        setPlanName,
        toastOpen,
        setToastOpen,
        drawerOpen,
        setDrawerOpen,
        externalSiteModal,
        setExternalSiteModal,
        resetExternalModal,
        innerWidth,
        setInnerWidth,
        globalError,
        setGlobalError,
        acknowledgmentModal,
        setAcknowledgmentModal,
        resetAcknowledgmentModal,
      }}
    >
      {children}
    </AppContext.Provider>
  );
};

export const useAppContext = () => {
  return useContext(AppContext);
};
