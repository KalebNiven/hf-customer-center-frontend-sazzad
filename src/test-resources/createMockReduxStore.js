import configureMockStore from "redux-mock-store";

export default () => {
  const mockStore = configureMockStore();

  var claims = {
    964311200: [],
    964311201: [],
    964311202: [],
    964311203: [],
    420000000696100: [],
  };
  var claimsList = [];
  var claimStatus = "";
  for (var member in claims) {
    claims[member].forEach((claim) => {
      if (claim.ClaimStatus === "Paid") {
        claimStatus = "Processed";
      } else {
        claimStatus = "Pending";
      }

      claimsList.push({
        id: claim.ClaimNo,
        claimId: claim.ClaimNo,
        memberId: claim.Member.MemberId,
        firstName: formatNameCapitalize(claim.Member.FirstName),
        lastName: formatNameCapitalize(claim.Member.LastName),
        provider: {
          firstName: claim.Provider.FirstName,
          lastName: claim.Provider.LastName,
        },
        serviceDate: formatDate(claim.ServiceFromDate),
        claimStatus: claimStatus,
        copayAmount: claim.CopayAmt,
        prepaidAmount: claim.PrepaidAmt,
      });
    });
  }

  const initialState = {
    claim: {
      loading: false,
      claimList: claimsList.map(
        (val) =>
          val && {
            ...val,
            memberName: val.firstName.concat(" ", val.lastName),
            providerName: val.provider.firstName.concat(
              " ",
              val.provider.lastName,
            ),
          },
      ),
    },
    customerInfo: {
      data: {
        hohPlans: [
          {
            CustomerId: "CUST-119586438",
            MemberId: "420000000696100",
            MembershipKey: "MHS|ENBMAS|420000000696100|QHP307|1|27960094",
            FirstName: "Sasha",
            LastName: "Buffin",
            DOB: "1979-09-05",
            age: 44,
            Gender: "M",
            subscriberId: "420000000696100",
            ZipCode: "11801",
            MembershipEffectiveDate: "2019-09-01",
            MembershipExpirationDate: "9999-12-31",
            CompanyNumber: "42",
            MembershipStatus: "active",
            PlanName: "Silver Leaf",
            RelationshipCode: "SELF",
            RelationshipTypeCode: "1",
            RelationshipRefCode: "18",
            BenefitPackage: "QHSV",
            GroupNumber: "QHP307",
            LOBCode: "QSVP",
            coverageStatus: "Enrolled",
            pcpId: "252519-A13",
            renewalDate: "2023-12-31",
            memberYear: "2024",
            CarrierCode: "",
          },
          {
            CustomerId: "CUST-119586438",
            MemberId: "964311200",
            MembershipKey: "MHS|MEMMAS|964311200|SCHP01|1",
            FirstName: "Sasha",
            LastName: "Buffin",
            DOB: "1979-09-05",
            age: 44,
            Gender: "M",
            subscriberId: "964311200",
            ZipCode: "11801",
            MembershipEffectiveDate: "2021-01-01",
            MembershipExpirationDate: "9999-12-31",
            CompanyNumber: "20",
            MembershipStatus: "active",
            PlanName: "CHILD HEALTH PLUS STATE PAY",
            RelationshipCode: "SELF",
            RelationshipTypeCode: "1",
            RelationshipRefCode: "18",
            BenefitPackage: "LI",
            GroupNumber: "SCHP01",
            LOBCode: null,
            coverageStatus: "Enrolled",
            pcpId: null,
            renewalDate: null,
            memberYear: "2024",
            CarrierCode: "",
          },
        ],
        previousPlans: [],
        customerId: "CUST-119586438",
        firstName: "Sasha",
        lastName: "Buffin",
        dob: "1979-09-05",
        membershipStatus: "active",
        email: "IPIATKOV@HEALTHFIRST.ORG",
        mobilePhone: "",
        emailStatus: false,
        phoneStatus: false,
        wantsMedicare: false,
        oktaId: "00u1255p1ddAVY84k0h8",
        accountStatus: "MEMBER",
        phoneNumber: "",
        planName: "Silver Leaf",
        companyCode: "42",
        sessLobCode: "QSVP",
        membershipEffectiveDate: "2019-09-01",
        membershipExpirationDate: "9999-12-31",
        memberId: "420000000696100",
        groupNumber: "QHP307",
        memberYear: "2024",
        language: "en",
        pcpId: "252519-A13",
        benefitPackage: "QHSV",
        zipcode: "11801",
        gender: "M",
        age: 44,
        csrf: "UECADejyDg1OdJrROZf1l6URyRxzI9nX1GsaRVmK",
        appVersion: "v1.1315.0",
        dependents: [
          {
            customerId: "CUST-118800909",
            memberId: "964311202",
            firstName: "SUZY",
            lastName: "BUFFIN",
            DOB: "2015-10-27",
            Age: 8,
            address: "",
            companyCode: "20",
            LobCode: "",
            groupNumber: "SCHP01",
            benefitPackage: "CHPS",
            MembershipKey: "964311202|SCHP01|1",
            Status: "active",
            disablePcpUpdate: false,
            RelationshipType: "CHILD",
            RelationshipTypeCode: "3",
            RelationshipRefCode: "14",
            CarrierCode: "",
            MembershipEffectiveDate: "2021-01-01",
            year: "2024",
            MembershipExpirationDate: "9999-12-31",
            BenefitPackageEffectiveDate: "2021-01-01",
            BenefitPackageExpirationDate: "9999-12-31",
            pcpId: null,
            planName: "Child Health Plus",
            renewalDate: "2023-12-31",
          },
          {
            customerId: "CUST-140493262",
            memberId: "964311201",
            firstName: "AMEL",
            lastName: "BUFFIN",
            DOB: "2020-06-17",
            Age: 3,
            address: "",
            companyCode: "20",
            LobCode: "",
            groupNumber: "SCHP01",
            benefitPackage: "CHPS",
            MembershipKey: "964311201|SCHP01|1",
            Status: "active",
            disablePcpUpdate: false,
            RelationshipType: "CHILD",
            RelationshipTypeCode: "3",
            RelationshipRefCode: "18",
            CarrierCode: "",
            MembershipEffectiveDate: "2021-01-01",
            year: "2024",
            MembershipExpirationDate: "9999-12-31",
            BenefitPackageEffectiveDate: "2021-01-01",
            BenefitPackageExpirationDate: "9999-12-31",
            pcpId: null,
            planName: "Child Health Plus",
            renewalDate: "2023-12-31",
          },
          {
            customerId: "CUST-118798257",
            memberId: "964311203",
            firstName: "ATHANASIOS",
            lastName: "BUFFIN",
            DOB: "2014-07-06",
            Age: 9,
            address: "",
            companyCode: "20",
            LobCode: "",
            groupNumber: "SCHP01",
            benefitPackage: "CHPS",
            MembershipKey: "964311203|SCHP01|1",
            Status: "active",
            disablePcpUpdate: false,
            RelationshipType: "CHILD",
            RelationshipTypeCode: "3",
            RelationshipRefCode: "14",
            CarrierCode: "",
            MembershipEffectiveDate: "2021-01-01",
            year: "2024",
            MembershipExpirationDate: "9999-12-31",
            BenefitPackageEffectiveDate: "2021-01-01",
            BenefitPackageExpirationDate: "9999-12-31",
            pcpId: null,
            planName: "Child Health Plus",
            renewalDate: "2023-12-31",
          },
          {
            customerId: "CUST-119590236",
            memberId: "420000000696101",
            firstName: "NORIKO",
            lastName: "BUFFIN",
            DOB: "1985-09-06",
            Age: 38,
            address: "",
            companyCode: "42",
            LobCode: "QSVP",
            groupNumber: "QHP307",
            benefitPackage: "QHSV",
            MembershipKey: "420000000696101|QHP307|1",
            Status: "active",
            disablePcpUpdate: false,
            RelationshipType: "SPOUSE",
            RelationshipTypeCode: "2",
            RelationshipRefCode: "01",
            CarrierCode: "",
            MembershipEffectiveDate: "2019-09-01",
            year: "2024",
            MembershipExpirationDate: "9999-12-31",
            BenefitPackageEffectiveDate: "2022-01-01",
            BenefitPackageExpirationDate: "9999-12-31",
            pcpId: null,
            planName: "Silver Leaf",
            renewalDate: "2023-12-31",
          },
        ],
      },
    },
    customerDemographicsInfo: {
      data: {
        hoh: [
          {
            memberId: "420000000696100",
            info: {
              addresses: [
                {
                  addressType: "Home",
                  addressPrecedence: 1,
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "0",
                  certifiedMailScore: "",
                  certifiedMailStatus: "I2",
                  latitude: "40.763593",
                  longitude: "-73.523059",
                  email: "SASHA.BUFFIN@HFTDM.COM",
                  phones: [
                    {
                      phoneType: "Home",
                      phoneNumber: "19312533389",
                    },
                    {
                      phoneType: "Other",
                      phoneNumber: "14322736639",
                    },
                  ],
                },
              ],
              contact: {
                addressType: "Home",
                prefix: "",
                firstName: "SASHA",
                middleName: "",
                lastName: "BUFFIN",
                suffix: "",
                fullName: "SASHA BUFFIN",
                contactType: "MEMBER",
                contactAddress: {
                  addressType: "Home",
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "0",
                  certifiedMailScore: "",
                  certifiedMailStatus: "I2",
                  latitude: "40.763593",
                  longitude: "-73.523059",
                  email: "SASHA.BUFFIN@HFTDM.COM",
                  phones: [
                    {
                      phoneType: "Home",
                      phoneNumber: "19312533389",
                    },
                    {
                      phoneType: "Other",
                      phoneNumber: "14322736639",
                    },
                  ],
                },
              },
            },
          },
          {
            memberId: "964311200",
            info: {
              addresses: [
                {
                  addressType: "Home",
                  addressPrecedence: 1,
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "0",
                  certifiedMailScore: "",
                  certifiedMailStatus: "I2",
                  latitude: "40.763593",
                  longitude: "-73.523059",
                  email: "SASHA.BUFFIN@HFTDM.COM",
                  phones: [
                    {
                      phoneType: "Home",
                      phoneNumber: "19312533389",
                    },
                    {
                      phoneType: "Other",
                      phoneNumber: "14322736639",
                    },
                  ],
                },
              ],
              contact: {
                addressType: "Home",
                prefix: "",
                firstName: "SASHA",
                middleName: "",
                lastName: "BUFFIN",
                suffix: "",
                fullName: "SASHA BUFFIN",
                contactType: "MEMBER",
                contactAddress: {
                  addressType: "Home",
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "0",
                  certifiedMailScore: "",
                  certifiedMailStatus: "I2",
                  latitude: "40.763593",
                  longitude: "-73.523059",
                  email: "SASHA.BUFFIN@HFTDM.COM",
                  phones: [
                    {
                      phoneType: "Home",
                      phoneNumber: "19312533389",
                    },
                    {
                      phoneType: "Other",
                      phoneNumber: "14322736639",
                    },
                  ],
                },
              },
            },
          },
        ],
        dependents: [
          {
            memberId: "964311202",
            info: {
              addresses: [
                {
                  addressType: "Home",
                  addressPrecedence: 1,
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.844966",
                  longitude: "-73.596804",
                  email: "",
                  phones: [],
                },
                {
                  addressType: "Responsible Party",
                  addressPrecedence: 2,
                  prefix: "",
                  firstName: "SASHA",
                  middleName: "",
                  lastName: "BUFFIN",
                  suffix: "",
                  fullName: "SASHA BUFFIN",
                  responsiblePartyType: "Miscellaneous Responsible Party",
                  addressLine1: "4496 MELVILLE STREET",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.840490",
                  longitude: "-73.633679",
                  email: "",
                  phones: [],
                },
              ],
              contact: {
                addressType: "Home",
                prefix: "",
                firstName: "SUZY",
                middleName: "LEVINGSTON",
                lastName: "BUFFIN",
                suffix: "",
                fullName: "SUZY LEVINGSTON BUFFIN",
                contactType: "MEMBER",
                contactAddress: {
                  addressType: "Home",
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.844966",
                  longitude: "-73.596804",
                  email: "",
                  phones: [],
                },
              },
            },
          },
          {
            memberId: "964311201",
            info: {
              addresses: [
                {
                  addressType: "Home",
                  addressPrecedence: 1,
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.844966",
                  longitude: "-73.596804",
                  email: "AMEL.BUFFIN@HFTDM.COM",
                  phones: [
                    {
                      phoneType: "Home",
                      phoneNumber: "19312533389",
                    },
                    {
                      phoneType: "Alt",
                      phoneNumber: "14322736639",
                    },
                    {
                      phoneType: "Other",
                      phoneNumber: "19312533389",
                    },
                  ],
                },
                {
                  addressType: "Responsible Party",
                  addressPrecedence: 2,
                  prefix: "",
                  firstName: "SASHA",
                  middleName: "",
                  lastName: "BUFFIN",
                  suffix: "",
                  fullName: "SASHA BUFFIN",
                  responsiblePartyType: "Miscellaneous Responsible Party",
                  addressLine1: "4496 MELVILLE STREET",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.840490",
                  longitude: "-73.633679",
                  email: "",
                  phones: [],
                },
              ],
              contact: {
                addressType: "Home",
                prefix: "",
                firstName: "AMEL",
                middleName: "MCDONALD",
                lastName: "BUFFIN",
                suffix: "",
                fullName: "AMEL MCDONALD BUFFIN",
                contactType: "MEMBER",
                contactAddress: {
                  addressType: "Home",
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.844966",
                  longitude: "-73.596804",
                  email: "AMEL.BUFFIN@HFTDM.COM",
                  phones: [
                    {
                      phoneType: "Home",
                      phoneNumber: "19312533389",
                    },
                    {
                      phoneType: "Alt",
                      phoneNumber: "14322736639",
                    },
                    {
                      phoneType: "Other",
                      phoneNumber: "19312533389",
                    },
                  ],
                },
              },
            },
          },
          {
            memberId: "964311203",
            info: {
              addresses: [
                {
                  addressType: "Home",
                  addressPrecedence: 1,
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.844966",
                  longitude: "-73.596804",
                  email: "",
                  phones: [],
                },
                {
                  addressType: "Responsible Party",
                  addressPrecedence: 2,
                  prefix: "",
                  firstName: "SASHA",
                  middleName: "",
                  lastName: "BUFFIN",
                  suffix: "",
                  fullName: "SASHA BUFFIN",
                  responsiblePartyType: "Miscellaneous Responsible Party",
                  addressLine1: "4496 MELVILLE STREET",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.840490",
                  longitude: "-73.633679",
                  email: "",
                  phones: [],
                },
              ],
              contact: {
                addressType: "Home",
                prefix: "",
                firstName: "ATHANASIOS",
                middleName: "",
                lastName: "BUFFIN",
                suffix: "",
                fullName: "ATHANASIOS BUFFIN",
                contactType: "MEMBER",
                contactAddress: {
                  addressType: "Home",
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.844966",
                  longitude: "-73.596804",
                  email: "",
                  phones: [],
                },
              },
            },
          },
          {
            memberId: "420000000696101",
            info: {
              addresses: [
                {
                  addressType: "Home",
                  addressPrecedence: 1,
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.844966",
                  longitude: "-73.596804",
                  email: "",
                  phones: [
                    {
                      phoneType: "Home",
                      phoneNumber: "19312533389",
                    },
                    {
                      phoneType: "Other",
                      phoneNumber: "14322736639",
                    },
                  ],
                },
              ],
              contact: {
                addressType: "Home",
                prefix: "",
                firstName: "NORIKO",
                middleName: "",
                lastName: "BUFFIN",
                suffix: "",
                fullName: "NORIKO BUFFIN",
                contactType: "MEMBER",
                contactAddress: {
                  addressType: "Home",
                  addressLine1: "1302 UPLAND AVENUE",
                  addressLine2: "",
                  city: "HICKSVILLE",
                  state: "NY",
                  zip: "11801",
                  zipExt: "",
                  county: "NASSAU",
                  country: "USA",
                  mailScore: "5",
                  certifiedMailScore: "",
                  certifiedMailStatus: "",
                  latitude: "40.844966",
                  longitude: "-73.596804",
                  email: "",
                  phones: [
                    {
                      phoneType: "Home",
                      phoneNumber: "19312533389",
                    },
                    {
                      phoneType: "Other",
                      phoneNumber: "14322736639",
                    },
                  ],
                },
              },
            },
          },
        ],
      },
    },
    submitClaimDetails: {
      loading: false,
    },
    submitAttestationAgreement: {
      loading: false,
    },
  };
  return mockStore(initialState);
};
