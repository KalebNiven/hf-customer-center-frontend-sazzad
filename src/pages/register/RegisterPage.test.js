import React from 'react'

import {fireEvent, render, screen} from '@testing-library/react'
import { Provider } from 'react-redux'

import {devTools} from "../../store/store";
import {AppContextProvider} from "../../AppContext";
import createMockReduxStore from "../../test-resources/createMockReduxStore";
import CreateAccount from "../../components/auth/registration/createAccount";
import {Router} from "react-router-dom";


describe('RegisterPage', () => {
    let store

    beforeEach(() => {
        process.env.MIX_APP_DOMAIN = 'https://place.com'
        window.analytics = { track: jest.fn()}
        store = createMockReduxStore()
    })

    it('renders correctly', () => {
        const {asFragment} = render(
            <AppContextProvider>
                    <Provider store={store}>
                        <Router history={{
                            push: () => {},
                            listen(listener) {},
                            location: {pathname: '/register'}
                        }}>
                            <CreateAccount ignoreSplit={true}/>
                        </Router>
                    </Provider>
            </AppContextProvider>)

        expect(asFragment()).toMatchSnapshot();
    })

    it('routes to /login when clicking healthfirst logo', async () => {
        const mockPush = jest.fn()

        render(
            <AppContextProvider>
                <Provider store={store}>
                    <Router history={{
                        push: mockPush,
                        listen(listener) {},
                        location: {pathname: '/register'}
                    }}>
                        <CreateAccount ignoreSplit={true}/>
                    </Router>
                </Provider>
            </AppContextProvider>)

        fireEvent.click(screen.getByTestId('healthfirst-logo'))

        expect(mockPush).toBeCalledWith('/login')
    })

    it('routes to /login when clicking back icon', async () => {
        const mockPush = jest.fn()

        render(
            <AppContextProvider>
                <Provider store={store}>
                    <Router history={{
                        push: mockPush,
                        listen(listener) {},
                        location: {pathname: '/register'}
                    }}>
                        <CreateAccount ignoreSplit={true}/>
                    </Router>
                </Provider>
            </AppContextProvider>)

        fireEvent.click(screen.getByTestId('back-icon'))

        expect(mockPush).toBeCalledWith('/login')
    })

    it('routes to /login when clicking back text', async () => {
        const mockPush = jest.fn()

        render(
            <AppContextProvider>
                <Provider store={store}>
                    <Router history={{
                        push: mockPush,
                        listen(listener) {},
                        location: {pathname: '/register'}
                    }}>
                        <CreateAccount ignoreSplit={true}/>
                    </Router>
                </Provider>
            </AppContextProvider>)

        fireEvent.click(screen.getByTestId('back-text'))

        expect(mockPush).toBeCalledWith('/login')
    })
});

