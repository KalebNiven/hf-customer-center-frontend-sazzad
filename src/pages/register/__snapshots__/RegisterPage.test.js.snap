// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Wrapper-sc-11yfjnx-0 eqQOMC"
  >
    <div
      class="Container-sc-11yfjnx-2 iWojAi"
    >
      <div
        class="LanguageSelectionWrapper-ev0rdi-0 kewMnM"
      >
        <ul
          class="List-kpclvo-0 edeZZK"
        >
          <li
            class="ListItem-kpclvo-1 dhkyhU"
          >
            <a
              href="https://member-rt.healthfirst.org/selectLanguage?selectedLang=en"
            >
              EN
            </a>
          </li>
          <li
            class="ListItem-kpclvo-1 dhkyhU"
          >
            <a
              href="https://es.member-rt.healthfirst.org/selectLanguage?selectedLang=es"
            >
              ES
            </a>
          </li>
          <li
            class="ListItem-kpclvo-1 dhkyhU"
          >
            <a
              href="https://zh.member-rt.healthfirst.org/selectLanguage?selectedLang=zh"
            >
              中文
            </a>
          </li>
        </ul>
      </div>
      <img
        class="LogoImg-ev0rdi-16 KsTGj"
        data-testid="healthfirst-logo"
        src="/react/images/simple_white.svg"
      />
      <div
        class="MemberCardsContainer-ev0rdi-12 szOgA"
      >
        <div
          class="MemberCard-ev0rdi-11 hfvcXC"
        >
          <div
            class="BackContainer-ev0rdi-1 FCSJF"
          >
            <img
              class="BackIcon-sc-1evaehd-22 wgEsB"
              data-testid="back-icon"
              src="/react/images/dark/ico-back.svg"
            />
            <p
              class="BackText-sc-1evaehd-23 eQfLwi"
              data-testid="back-text"
            >
              Back
            </p>
          </div>
          <div
            class="HeaderText-ev0rdi-6 hJmGCV"
          >
            What is your current membership status with Healthfirst?
          </div>
          <div
            class="SubHeaderText-ev0rdi-7 iRCOd"
          >
            Interested in Healthfirst plans? 
            <a
              class="LearnLink-ev0rdi-8 ePFGbr"
              href="https://www2.healthfirst.org/shoppers"
              target="_shoppers"
            >
              Learn More. 
            </a>
          </div>
          <div
            class="RadioButtonContainer-ev0rdi-10 lmbkrE"
          >
            <img
              class="RadioImg-ev0rdi-13 jDcfhX"
              src="/react/images/icn-radio-active.svg"
            />
            <div>
              <div
                class="TextHeader-ev0rdi-2 jjeqpx mt-0"
              >
                Active Healthfirst Member
              </div>
              <div
                class="pt-2"
              >
                <div
                  class="TextWrapper-ev0rdi-9 eUOwzR"
                >
                  I am currently enrolled in a Healthfirst plan and have a Member ID, but I do not have an online account.
                </div>
              </div>
            </div>
          </div>
          <div
            class="RadioButtonContainer-ev0rdi-10 cmjfcG"
          >
            <img
              class="RadioImg-ev0rdi-13 jDcfhX"
              src="/react/images/icn-radio-inactive.svg"
            />
            <div>
              <div
                class="TextHeader-ev0rdi-2 jjeqpx mt-0"
              >
                I Need to Activate My Coverage
              </div>
              <div
                class="pt-2"
              >
                <div
                  class="TextWrapper-ev0rdi-9 eUOwzR"
                >
                  I am enrolled in a Healthfirst plan, but my coverage isn't active as I need to make my first premium payment.
                </div>
              </div>
            </div>
          </div>
          <button
            class="StyledButton-sc-1evaehd-13 SubmitButton-ev0rdi-14 guCLPh iRbqOd"
          >
            Continue
          </button>
           
        </div>
        <div
          class="FooterBox-ev0rdi-4 cMFAOF"
        >
          <div
            class="FooterBoxContent-ev0rdi-5 XycUz"
          >
            If you have any questions, please call the Member Services number on your Member ID card or visit us at
            <a
              class="BoldText-ev0rdi-3 kVcdBP"
              href="https://healthfirst.org/contact"
              target="healthfirstContactus"
            >
               healthfirst.org/contact.
            </a>
          </div>
        </div>
      </div>
      <div
        class="FooterMenuWrapper-sc-11yfjnx-19 cpTGsJ"
      >
        <div
          class="Wrapper-sc-1t67vgj-0"
        >
          <ul
            class="List-sc-1t67vgj-1 zobnX"
          >
            <li
              class="ListItem-sc-1t67vgj-2 elfIE"
            >
              <a
                href="https://healthfirst.org/contact"
              >
                Contact Us
              </a>
            </li>
            <li
              class="ListItem-sc-1t67vgj-2 elfIE"
            >
              <a
                href="https://assets.healthfirst.org/pdf_bb1adede22e833c08a2d763580d25ab3?v=0711074100"
              >
                Terms & Conditions
              </a>
            </li>
            <li
              class="ListItem-sc-1t67vgj-2 elfIE"
            >
              <a
                href="http://assets.healthfirst.org/api/pdf?id=pdf_267ff1120f&key=2a057fc3de988ef49277dcc39931e7444611493e"
              >
                Web Privacy Statement
              </a>
            </li>
            <li
              class="ListItem-sc-1t67vgj-2 elfIE"
            >
              <a
                href="http://assets.healthfirst.org/api/pdf?id=pdf_53b845cf68&key=990ce7f9a53a3c552ac2fa2ca16f0dd7a9cdac0b&_ga=2.262813516.1197467741.1496254267-1737878043.1427400676"
              >
                New York Privacy Notice
              </a>
            </li>
            <li
              class="ListItem-sc-1t67vgj-2 elfIE"
            >
              <a
                href="http://assets.healthfirst.org/api/pdf?id=pdf_ffc2ec5002&key=747d08e9f76a07931acd9d93fd2370fac86900e7&_ga=2.259676362.1197467741.1496254267-1737878043.1427400676"
              >
                HIPAA Privacy Notice
              </a>
            </li>
          </ul>
          <div
            class="Copyright-sc-1t67vgj-3 blZWFD"
          >
            © 2024 Healthfirst
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
